### PR TITLE
Include token balances on UTxO sum response

### PR DESCRIPTION
**Pull request by pedromtcosta**
# Abstract
This PR intends to add the total token balances to the `/txs/utxoSumForAddresses`.

# Implementation
The implementation consists of adding another query to `askUtxoSumForAddresses` that works very similarly to the regular UTxO summing query, with the only differences that we also make a join to the `ma_tx_out` table to get the outputs for assets other than ADA and sum the `quantity` column from that table grouped by the asset's name and policy. Because `utxo_view` already filters the unspent TX outputs, we don't have to do it again. We then simply change the the function to return the ADA sum in a new field named `totalAda` (`sum` was kept for backward compatibility), and the token balances in a new `tokensBalance` field.

### CODE REVIEW

File: src/services/utxoSumForAddress.ts [hunk 1]
```diff
@@ -1,25 +1,52 @@
...
+      FROM "public"."utxo_view"
+      WHERE "public"."utxo_view"."address" = any(($1)::varchar array)
+      GROUP BY "public"."utxo_view"."value"
+      ORDER BY "public"."utxo_view"."value" ASC
+    `;
+
+    const tokensQuery = `
+      SELECT SUM(ma_utxo.quantity) amount,
+        encode(ma_utxo.policy, 'hex') as policy,
+        convert_from(ma_utxo.name, 'UTF8') as name
```
Comment (yushih) [inline src/services/utxoSumForAddress.ts:18]: Not sure if the `name` is encoded as utf8. It seems to me this is actually interpreted as ASCII code in Yoroi (https://github.com/Emurgo/yoroi-frontend/blob/develop/packages/yoroi-extension/app/stores/stateless/tokenHelpers.js#L29). I'd suggest just return the hex string. Please correct me @vsubhuman .

File: src/services/utxoSumForAddress.ts [hunk 2]
```diff
@@ -1,25 +1,52 @@
...
   // TODO: support for payment keys
-    const sqlQuery = `
-    SELECT "public"."utxo_view"."value" AS "value"
-    FROM "public"."utxo_view"
-    WHERE "public"."utxo_view"."address" = any(($1)::varchar array)
-    GROUP BY "public"."utxo_view"."value"
-    ORDER BY "public"."utxo_view"."value" ASC
-  `;
+    const adaQuery = `
+      SELECT "public"."utxo_view"."value" AS "value"
```
Comment (yushih) [inline src/services/utxoSumForAddress.ts:8]: This original query looks strange to me. Shouldn't it sum the value but leave the "group by" and "order by" out? cc @vsubhuman

**Review by yushih — COMMENTED** [review]
The added `tokensQuery`

**Comment by yushih** [comment]
Please update the API doc in README, too.

File: src/services/utxoSumForAddress.ts (hunk 2 above)
Comment (pedromtcosta) [reply src/services/utxoSumForAddress.ts:8]: The `group by` and `order by` on the original query indeed have no purpose and they can be safely removed, so I did it

Comment (yushih) [reply src/services/utxoSumForAddress.ts:8]: Also, should it be `SELECT SUM(...) AS "value"`?

Comment (pedromtcosta) [reply src/services/utxoSumForAddress.ts]: Definitely should, totally missed that, sorry! Fixed in the latest commit

**Review by yushih — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -234,11 +234,17 @@ We recommend querying using payment key hashes (`addr_vkh`) when possible (other
...
   ```js
   {
-    sum: ?string
+    sum: ?string,
+    totalAda: ?string,
+    tokensBalance: [
+      amount: string,
+      name: string,
+      policy: string
+    ]
```
Comment (vsubhuman) [inline README.md:247]: Discussed with @pedromtcosta :
1. Remove `totalAda`
2. Replace `name` and `policy` with `assetId: "{policy}.{name}"` to match other API parts

Comment (pedromtcosta) [reply README.md]: Done

**Review by vsubhuman — APPROVED** [review]

**Pull request merged**